reader-v2.go

```diff
@@ -180,6 +180,36 @@ func (r *decReaderV20) Read(p []byte) (n int, err error) {
 	return n, nil
 }
 
+// decryptBufferV20 will append plaintext to dst and return the result.
+func decryptBufferV20(dst, src []byte, config *Config) ([]byte, error) {
+	ad, err := newAuthDecV20(config)
+	if err != nil {
+		return nil, err
+	}
+	for len(src) > 0 {
+		buffer := packageV20(src)
+		// Truncate to max package size
+		if len(buffer) > maxPackageSize {
+			buffer = buffer[:maxPackageSize]
+		}
+
+		// Make space in dst
+		payloadLen := buffer.Header().Length()
+		dst = append(dst, make([]byte, payloadLen)...)
```

Comment:
Shouldn't we check whether `dst` has enough `cap` and avoid the allocation if so?

Reply:
@aead I seem to remember the compiler being clever enough to pick this up, so it never allocates (unless of course dst is too small). I will double check though.

Reply:
We do avoid the zeroing, though.

Reply:
Yeah, it just does `(d:\minio\sio\reader-v2.go:198)	CALL	runtime.memclrNoHeapPointers(SB)` on the dst slice. No allocs for the `make`. But hey let's get rid of the zeroing.

Reply:
@aead Updated.